NetCasbin/Enforcer.cs

```diff
@@ -631,12 +535,16 @@ private bool Enforce(IReadOnlyList<object> requestValues, ICollection<IEnumerabl
                 IReadOnlyList<string> policyValues = Enumerable.Repeat(string.Empty, policyTokenCount).ToArray();
                 ExpressionHandler.SetPolicyParameters(policyValues);
                 var nowEffect = GetEffect(ExpressionHandler.Invoke(expressionString, requestValues));
-                finalResult = Effector.MergeEffects(effect, new[] { nowEffect }, null, out hitPolicyIndex);
-            }
 
-            if (explain && hitPolicyIndex is not -1)
-            {
-                explains.Add(policyList[hitPolicyIndex]);
+                if (effector.TryChain(nowEffect))
```

Comment:
use `TryChain` to replace `MergeEffects`? I do not quite understand this method, but it should be right because all the tests passed.